### PR TITLE
fix: Multiple virtual cluster issues

### PR DESCRIPTION
**Pull request by shivani170**
# Description

Multi fix
         1. Three dot menu should be wrap with a div
         2. Menu opening position
         3. Menu pt-4 pb-4
         4. For virtual env show 'Last deployed on {envName}' instead of 'Active on {envName}' //Missed sequential cd case
         5. Generalize logic for pre/post config 
         6. Preview title check modified

Fixes [#AB3642](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=3642)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/service.ts [hunk 1]
```diff
@@ -374,6 +374,7 @@ function cdMaterialListModal(
             artifactStatus: artifactStatusValue,
             userApprovalMetadata: material.userApprovalMetadata,
             triggeredBy: material.triggeredBy,
+            isVirtualEnvironment: material?.isVirtualEnvironment,
```
Comment (vivek-devtron) [inline src/components/app/service.ts:377]: Please remove the optional check

File: src/components/v2/appDetails/sourceInfo/sourceInfo.css [hunk 2]
```diff
@@ -26,6 +26,18 @@
     } */
 }
 
+.helm-delete-pop-up{
+  position: fixed;
+  left: unset;
+  top: 130px;
+  opacity: 1;
+  right: 20px;
+  background-color: var(--N000);
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/sourceInfo/sourceInfo.css:35]: please reduce the CSS

File: src/components/app/service.ts (hunk 1 above)
Comment (shivani170) [reply src/components/app/service.ts:377]: Once i will get the BE key will remove

Comment (shivani170) [reply src/components/app/service.ts:377]: Done

File: src/components/v2/appDetails/sourceInfo/sourceInfo.css (hunk 2 above)
Comment (shivani170) [reply src/components/v2/appDetails/sourceInfo/sourceInfo.css]: Done

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**